client/src/app/_components/shared/product-search/product-search.component.ts

```
@@ -0,0 +1,145 @@
+
+  @HostListener('keydown', ['$event'])
+  handleKeyboardEvent(event: KeyboardEvent): void {
+    if (!this.mostrarResultados()) return;
+
+    const productos = this.productosEncontrados();
+    const currentIndex = this.selectedProductIndex();
+
+    switch (event.key) {
+      case 'ArrowDown':
+        event.preventDefault();
+        const nextIndex = currentIndex < productos.length - 1 ? currentIndex + 1 : 0;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)